src/world.cpp

```diff
@@ -2208,7 +2208,7 @@ void World_GenRoom(struct room_s *room, class VT_Level *tr)
     room->alternate_room = NULL;
     room->base_room = NULL;
 
-    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count))
+    if((tr_room->alternate_room >= 0) && ((uint32_t)tr_room->alternate_room < tr->rooms_count) && (room->id < tr_room->alternate_room))
```

Comment:
But how TR1 determined which room to begin with? For example it can start in room 1 and then swap to room 2. But at the same time there are examples when it starts in room 2 and swaps to room 1. How it finds out which room should be the first?